tests/moduletool/test_install.py

```diff
@@ -462,13 +476,19 @@ def test_project_install_modules_cache_invalid(
         autostd=False,
         install_project=False,
         python_package_sources=[index.url, local_module_package_index],
+        # don't include preinstalled module to make sure to trigger the v1 loading error
+        python_requires=[Requirement.parse(module.ModuleV2Source.get_python_package_name(new_module_name))],
     )
 
     if not preinstall_v2:
         # populate project.modules[module_name]
-        module.Project.get().get_module(module_name, install=False, allow_v1=True)
+        module.Project.get().get_module(module_name, allow_v1=True)
 
     os.chdir(module.Project.get().path)
+    # TODO: this fails for the v1 preinstalled test because the v2 modules now get installed before starting the import based
+    # install flow for v1 modules. This means the v1 error is very much an edge case that can only occur when a v2 module is
+    # a transient dependency of a v1 module and this v2 module promotes another v1 to v2. Not sure whether the test should cover
+    # this or whether we should just drop the v1 preinstalled test.
```

Comment:
I'd forgotten about this. I'll still need to fix this but I'll open the PR for review in the meantime anyway because the faster it's merged in the faster the build timeouts will disappear. My fix here should only affect this one test case so you should be able to review all else already.

Reply:
Perhaps this can be tested by hand once and then left behind?

Reply:
Now they've started failing again (not sure how that's possible since I don't recall making any changes since they succeeded) so I'll have to look at it again tomorrow in any case.

Reply:
Stupid mistake, fixed.